mlrun/utils/notifications/notification/base.py

```diff
@@ -59,6 +69,13 @@ def load_notification(
     ) -> None:
         self.params = params or {}
 
+    @classmethod
+    def enrich_default_params(cls, params: dict, default_params: dict = None) -> dict:
+        default_params = default_params or {}
+        returned_params = deepcopy(default_params)
+        returned_params.update(params)
+        return returned_params
+
```

Comment:
Since each notification class calls the `enrich_default_params`, I'm not sure `get_notifications_default_params` is needed.
Each class can have its own `default_params` method which implements the default params however it needs, and the base `enrich_default_params` can call the inherited `default_params` of each notification object.

Reply:
You definitely right, but because for the current version the notifications are under the client code(`mlrun/mlrun/utils/notifications/notification`) and for mail notification for example we need some server code for the default params I choose to do it on the server side code.
After implementing [Make running notification send from the server side](https://iguazio.atlassian.net/browse/ML-8069) we can move the all notifications classes to the server side code and then I will also make the `default_params` inside the notification itself by the type of the notification.
For now, maybe I will add TODO about it. WDYT?

Reply:
yes add a TODO that would be best